docker-cpp-demos/amazonlinux-docker/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Build docker with
+# docker build -t kinesis-video-producer-sdk-cpp-amazon-linux .
+#
+FROM amazonlinux:latest
+RUN yum update -y && \
+	yum upgrade -y && \
+	yum install -y autoconf && \
+	yum install -y automake && \
+	yum install -y bison && \
+	yum install -y bzip2 && \
+	yum install -y cmake && \
+	yum install -y curl && \
+	yum install -y diffutils && \
+	yum install -y flex  && \
+	yum install -y gcc  && \
+	yum install -y gcc-c++ && \
+	yum install -y git && \
+	yum install -y gmp-devel && \
+	yum install -y libffi && \
+	yum install -y libffi-devel && \
+	yum install -y libmpc-devel && \
+	yum install -y libtool && \
+	yum install -y m4 && \
+	yum install -y mpfr-devel && \
+	yum install -y pkgconfig && \
+	yum install -y vim && \
+	yum install -y wget && \
+	yum install -y make && \
+	yum install -y xz && \
+	yum install -y gstreamer*
+
+RUN yum install -y libcurl-devel 
+
+RUN wget https://cmake.org/files/v3.12/cmake-3.12.3.tar.gz && \
+    tar zxvf cmake-3.* && \
+    rm cmake-3.12.3.tar.gz
+
+WORKDIR cmake-3.12.3/
+RUN ./bootstrap
+RUN make -j4
+RUN make install
+
+WORKDIR /opt/
+RUN git clone https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp.git
+WORKDIR /opt/amazon-kinesis-video-streams-producer-sdk-cpp/
+RUN git submodule update --init
+WORKDIR /opt/amazon-kinesis-video-streams-producer-sdk-cpp/build/
+RUN cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
+RUN make
+WORKDIR /opt/amazon-kinesis-video-streams-producer-sdk-cpp/build/
+ENV LD_LIBRARY_PATH=/opt/amazon-kinesis-video-streams-producer-sdk-cpp/open-source/local/lib
+ENV GST_PLUGIN_PATH=/opt/amazon-kinesis-video-streams-producer-sdk-cpp/build/:$GST_PLUGIN_PATH
```

Comment:
Probably delete the unnecessary artifacts after the build steps to reduce the docker image.

Reply:
Not sure if there are any.

Reply:
Yep, there are, such as the source codes, the headers, etc. But, feel free to ignore it. It's really not a big deal since this is just a demo.